app/models/miq_region_remote.rb

```diff
@@ -103,8 +103,7 @@ def self.with_remote_connection(host, port, username, password, database, adapte
         :database        => database,
         :connect_timeout => connect_timeout
       }.delete_blanks)
-      conn = pool.connection
-      yield conn
+      pool.with_connection { |conn| yield conn }
```

Comment:
per rubocop, does it make sense to:

```ruby
 def self.with_remote_connection(...,&block)
```

and then use here?

```suggestion
      pool.with_connection { |conn| block.yield(conn) }
```

Reply:
Sure, I'll ignore the bot complaining about argument length 😉

Reply:
```ruby
vmdb(dev)> MiqRegionRemote.with_remote_connection('localhost', 5432, 'root', 'XXXXXX', 'vmdb_demo_env_03132025', 'postgresql') { |conn| puts conn.raw_connection.db; conn.select_one("select count(*) from containers;")}
vmdb_demo_env_03132025
   (3.5ms)  select count(*) from containers;
=> {"count"=>1869}
```

```ruby
vmdb(dev)> MiqRegionRemote.with_remote_connection('localhost', 5432, 'root', 'XXXXXXXX', 'vmdb_production', 'postgresql') { |conn| puts conn.raw_connection.db; conn.select_one("select count(*) from containers;")}
vmdb_production
   (3.9ms)  select count(*) from containers;
=> {"count"=>0}
```

Reply:
lol

```
❗ - [Line 106](https://github.com/jrafanie/manageiq/blob/4bd383a91555882858d90fca58b4b604562fa750/app/models/miq_region_remote.rb#L106), Col 37 - Performance/RedundantBlockCall - Use yield instead of block.call.
```